From 0561a11c5e6fe3dc45d18540e853e56dce745fe2 Mon Sep 17 00:00:00 2001
From: Myy <myy@miouyouyou.fr>
Date: Sun, 13 Nov 2016 08:16:32 +0000
Subject: [PATCH 02/12] Integrate the Mali GPU address to the rk3288 and
 rk3288-miqi DTS

This adds the Mali GPU address in the Rockchip 3288 device tree
definition, and adds a reference to this definition in the miqi DTS
file.

The MiQi modifications were tested with the Mali Kernel-Space driver,
version r14p0.

You'll still need to include Mali driver support to the kernel, though.

Signed-off-by: Myy <myy@miouyouyou.fr>
---
 arch/arm/boot/dts/rk3288-miqi.dts |  5 +++++
 arch/arm/boot/dts/rk3288.dtsi     | 27 +++++++++++++++++++++++++++
 2 files changed, 32 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288-miqi.dts b/arch/arm/boot/dts/rk3288-miqi.dts
index 21326f3..cc42be5 100644
--- a/arch/arm/boot/dts/rk3288-miqi.dts
+++ b/arch/arm/boot/dts/rk3288-miqi.dts
@@ -470,3 +470,8 @@
 &wdt {
 	status = "okay";
 };
+
+&gpu {
+  mali-supply = <&vdd_gpu>;
+  status = "okay";
+};
diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index df8a0db..3bf38f1 100644
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -99,6 +99,8 @@
 			resets = <&cru SRST_CORE0>;
 			operating-points = <
 				/* KHz    uV */
+        1800000 1350000
+        1704000 1350000
 				1608000 1350000
 				1512000 1300000
 				1416000 1200000
@@ -177,6 +179,31 @@
 		};
 	};
 
+	gpu: mali@ffa30000 {
+		compatible = "arm,malit764",
+			     "arm,malit76x",
+			     "arm,malit7xx",
+			     "arm,mali-midgard";
+		reg = <0xffa30000 0x10000>;
+		interrupts = <GIC_SPI 6 IRQ_TYPE_LEVEL_HIGH>,
+			     <GIC_SPI 7 IRQ_TYPE_LEVEL_HIGH>,
+			     <GIC_SPI 8 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "JOB", "MMU", "GPU";
+		clocks = <&cru ACLK_GPU>;
+		clock-names = "clk_mali";
+		operating-points = <
+			/* KHz uV */
+			100000 950000
+			200000 950000
+			300000 1000000
+			400000 1100000
+			500000 1200000
+			600000 1250000
+		>;
+		power-domains = <&power RK3288_PD_GPU>;
+		status = "disabled";
+	};
+
 	reserved-memory {
 		#address-cells = <1>;
 		#size-cells = <1>;
-- 
2.10.2

